src/merkle/nibble.rs

```diff
@@ -204,6 +204,28 @@ pub fn encode(vec: NibbleSlice, typ: NibbleType, s: &mut RlpStream) {
     s.append(&ret);
 }
 
+#[derive(Debug)]
+pub struct Entry<V> {
+    pub key: NibbleVec,
+    pub value: V,
+}
+
+impl<V> Entry<V> {
+    pub fn new(key: NibbleVec, value: V) -> Self {
+        Self { key, value }
+    }
+
+    pub fn and_then<U, F>(self, f: F) -> Option<Entry<U>>
```

Comment:
`and_then` is used for "failable" types and maps their "valid" value to same type, with possible error propagation.
Like in `Option<T> -> Option<U>` or  `Result<T,_> -> Result<U,_>`

But you work with "unfailable" `Entry`  and map it to `Option`. Maybe more correct semantic would be `try_map`

Like in:
https://doc.rust-lang.org/std/primitive.array.html#method.try_map